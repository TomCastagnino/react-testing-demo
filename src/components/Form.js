import React, { Component } from 'react';

class Form extends Component {

    constructor(props) {
        super(props);
        this.state = {
            firstNumber: '',
            secondNumber: '',
            componentState: 'default'
        }
        this.handleFirstNumber = this.handleFirstNumber.bind(this);
        this.handleSecondNumber = this.handleSecondNumber.bind(this);
        this.displayResult = this.displayResult.bind(this);
        // this.handleAdd = this.handleAdd.bind(this);
        this.handleSubstract = this.handleSubstract.bind(this);
    }

    componentDidMount() {
        this.setState( {componentState: 'mounted'} );
    }

    handleFirstNumber(event) {
        this.setState( {firstNumber: event.target.value} );
    }

    handleSecondNumber(event) {
        this.setState( {secondNumber: event.target.value} );
    }

    handleAdd() {
        const { firstNumber, secondNumber } = this.state;
        this.displayResult(parseInt(firstNumber) + parseInt(secondNumber));
    }

    handleSubstract() {
        const { firstNumber, secondNumber } = this.state;
        this.displayResult(parseInt(firstNumber) - parseInt(secondNumber));
    }

    displayResult(result) {
        alert(result);
    }

    render() {
        const { firstNumber, secondNumber } = this.state;
        const { operator } = this.props;
        return (
            <form className='form-group'>
                <fieldset className='form-group'>
                    <label>
                        First Number:
                    </label>
                    <input type='text' id='number1' className='form-input' value={firstNumber} onChange={this.handleFirstNumber} />
                </fieldset>
                <fieldset className='form-group'>
                    <label>
                        Second Number:
                    </label>
                    <input type='text' id='number2' className='form-input' value={secondNumber} onChange={this.handleSecondNumber} />
                </fieldset>
                <div className='form-group'>
                    {operator === '+' &&
                        <button id='formButtonAdd' className='btn' type='button' onClick={() => this.handleAdd()}>
                            Add
                        </button>
                    }
                    {operator === '-' &&
                        <button id='formButtonSubstract' className='btn' type='button' onClick={this.handleSubstract}>
                            Substract
                        </button>
                    }
                </div>
            </form>
        );
    }
}

export default Form;