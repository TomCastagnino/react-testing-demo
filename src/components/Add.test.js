import React from 'react';
import { shallow, mount } from 'enzyme';
import Add from './Add';
import Form from './Form';

let wrapper;

beforeEach(() => {
    wrapper = shallow(<Add />);
});

describe('<Add /> rendering', () => {
    it('should render one <h1>', () => {
        expect(wrapper.find('h1')).toHaveLength(1);
    });
    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should render one <Form>', () => {
        expect(wrapper.find(Form)).toHaveLength(1);
    });
    it('should render two labels', () => {
        wrapper = mount(<Add />);
        expect(wrapper.find('label')).toHaveLength(2);
        wrapper.unmount();
    });
});