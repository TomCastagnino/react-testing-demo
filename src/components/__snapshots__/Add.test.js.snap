// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Add /> rendering should match snapshot 1`] = `
<div>
  <h1>
    Add Function
  </h1>
  <Form
    operator="+"
  />
</div>
`;
